src/python/zquantum/qcbm/ansatz_utils.py

```diff
@@ -55,7 +81,75 @@ def get_entangling_layer_line_topology(
     """
     assert params.shape[0] == n_qubits - 1
 
+    line_graph = np.zeros((n_qubits, n_qubits))
+    for i in range(n_qubits - 1):
+        line_graph[i][i + 1] = 1
+    return get_entangling_layer_graph_topology(
+        params, n_qubits, entangling_gate, line_graph
+    )
+
+
+def get_entangling_layer_star_topology(
+    params: np.ndarray, n_qubits: int, entangling_gate: GatePrototype, center_qubit: int
+) -> Circuit:
+    """Builds a circuit representing an entangling layer according to the star topology.
+
+    Args:
+        params (numpy.array): parameters of the circuit.
+        n_qubits (int): number of qubits in the circuit.
+        entangling_gate (str): gate specification for the entangling layer.
+        center_qubit (int): the center qubit of the star topology.
+    """
+    assert center_qubit < n_qubits
+    assert params.shape[0] == n_qubits - 1
+
+    star_graph = np.zeros((n_qubits, n_qubits))
+    for i in range(0, n_qubits):
+        star_graph[i][center_qubit] = 1
+    return get_entangling_layer_graph_topology(
+        params, n_qubits, entangling_gate, star_graph
+    )
+
+
+def get_entangling_layer_graph_topology(
+    params: np.ndarray,
+    n_qubits: int,
+    entangling_gate: GatePrototype,
+    adjacency_matrix: np.ndarray,
+) -> Circuit:
+    """Builds a circuit representing an entangling layer according to a general graph topology.
+
+    Args:
+        params: parameters of the circuit.
+        n_qubits: number of qubits in the circuit.
+        entangling_gate: gate specification for the entangling layer.
+        adjacency_matrix: adjacency matrix for the entangling layer.
+    """
+    assert adjacency_matrix.shape[0] == adjacency_matrix.shape[1] == n_qubits
+
     circuit = Circuit()
+    i = 0
     for qubit1_index in range(n_qubits - 1):
-        circuit += entangling_gate(params[qubit1_index])(qubit1_index, qubit1_index + 1)
+        for qubit2_index in range(qubit1_index + 1, n_qubits):
+            if (
+                adjacency_matrix[qubit1_index][qubit2_index]
+                or adjacency_matrix[qubit2_index][qubit1_index]
+            ):
+                circuit += entangling_gate(params[i])(qubit1_index, qubit2_index)
+                i += 1
     return circuit
+
+
+def adjacency_list_to_matrix(n_qubits: int, adj_list: np.ndarray):
```

Comment:
Lacks return type.
```suggestion
def adjacency_list_to_matrix(n_qubits: int, adj_list: np.ndarray) -> np.ndarray:
```
You can simply accept this suggestion in the GitHub UI by clicking "Commit suggestion".